circuit-examples/examples/halo2_collatz.rs

```diff
@@ -0,0 +1,46 @@
+// Given `start_number` and `output_number`, this circuit checks that applying the
+// Collatz map `collatz_iterations` many times results in `output_number`.
+//
+// The following should succeed with 6 collatz iterations:
+// `cargo run --release --example halo2_collatz_runtime 27 94`
+//
+use ark_bn254::Fr;
+use rcc::traits::BoolWire;
+use rcc_halo2::builder::{H2Wire as W, *};
+
+#[main_component]
+fn my_circuit() {
+    let collatz_iterations = 6;
+
+    let start_number = input_wire("start_number");
+    let output_number = input_wire("output_number");
+    output_number.declare_public("output_number");
+
+    let mut current_number = start_number;
+
+    let builder = start_number.builder();
+    let two = builder.new_constant_wire(Fr::from(2));
+    let two_inverse = builder.inv_or_panic(two);
```

Comment:
The computes the inverse and constrains in the circuit, which is unnecessary. Should be the following

```rust
let two_inverse = builder.new_constant_wire(Fr::from(2).inverse().unwrap())
```

Reply:
Actually we could just forego instantiating a wire here and do
```
let two_inverse = Fr::from(2).inverse().unwrap();
```

Reply:
Ah, nice!